aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -0,0 +1,149 @@
+import base64
+import json
+import re
+import zlib
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+from aws_lambda_powertools.shared.json_encoder import Encoder
+from aws_lambda_powertools.utilities.data_classes import ALBEvent, APIGatewayProxyEvent, APIGatewayProxyEventV2
+from aws_lambda_powertools.utilities.data_classes.common import BaseProxyEvent
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+class ProxyEventType(Enum):
+    http_api_v1 = "APIGatewayProxyEvent"
+    http_api_v2 = "APIGatewayProxyEventV2"
+    alb_event = "ALBEvent"
+    api_gateway = http_api_v1
+
+
+class Route:
+    def __init__(
+        self, method: str, rule: Any, func: Callable, cors: bool, compress: bool, cache_control: Optional[str]
+    ):
+        self.method = method.upper()
+        self.rule = rule
+        self.func = func
+        self.cors = cors
+        self.compress = compress
+        self.cache_control = cache_control
+
+
+class Response:
+    def __init__(self, status_code: int, content_type: str, body: Union[str, bytes], headers: Dict = None):
+        self.status_code = status_code
+        self.body = body
+        self.base64_encoded = False
+        self.headers: Dict = headers or {}
+        self.headers.setdefault("Content-Type", content_type)
+
+    def add_cors(self, method: str):
+        self.headers["Access-Control-Allow-Origin"] = "*"
+        self.headers["Access-Control-Allow-Methods"] = method
+        self.headers["Access-Control-Allow-Credentials"] = "true"
```

Comment:
Let's allow customers to specify a more strict CORS header as this is overly permissive. If they don't specify.

For example, `Allow-Origin` being `*`, and `Allow-Methods` isn't allowing or testing for pre-flight CORS (`OPTIONS`) and it'll fail.

Reply:
It's also missing [Allow-Headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers) - For that, you'd want to have a default of `Content-Type`.

Reply:
Ok @heitorlessa i will further in to that. I was thinking we could provide a simple default. And then if they want more flexibility, then just use the `Response` class.

Any ideas on the DX for this extra settings?

Reply:
I will have a look at the DX for [Chalice and CORS](https://aws.github.io/chalice/tutorials/basicrestapi.html#cors-support) and other frameworks. Seems like the lightweight flag would just add the `OPTIONS` route and “*” defaults.

Reply:
@heitorlessa I have added a more complete implementation (looks like CORS is a very large part of this which is a bummer, but for those that struggle with this, I guess this is a bonus)